Sources/SwiftSyntax/SyntaxData.swift

```diff
@@ -160,12 +160,45 @@ struct AbsoluteRawSyntax {
   }
 }
 
+/// Indirect wrapper for a `Syntax` node to avoid cyclic inclusion of the 
+/// `Syntax` struct in `SyntaxData`
+class SyntaxBox: CustomStringConvertible, 
```

Comment:
This might work nicely as a `@Boxed<T>` property wrapper. You wouldn't need to conform it to CustomStringConvertible or add the `write(to:)` method. What do you think?

Reply:
Talked offline, there are some downsides to a property wrapper (namely, inability to make the box itself Optional), so we're sticking with SyntaxBox.